src/governance/LendingTermOnboarding.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {Governor, IGovernor} from "@openzeppelin/contracts/governance/Governor.sol";
+
+import {Core} from "@src/core/Core.sol";
+import {CoreRoles} from "@src/core/CoreRoles.sol";
+import {GuildToken} from "@src/tokens/GuildToken.sol";
+import {LendingTerm} from "@src/loan/LendingTerm.sol";
+import {VoltGovernor} from "@src/governance/VoltGovernor.sol";
+
+/// @notice Utils to onboard a LendingTerm. Also acts as a LendingTerm factory.
+/// This contract acts as Governor, but users cannot queue arbitrary proposals,
+/// they can only queue LendingTerm onboarding proposals.
+/// When a vote is successful, the LendingTerm onboarding is queued in the Timelock,
+/// where CREDIT holders can veto the onboarding.
+/// Only terms that have been deployed through this factory can be onboarded.
+/// A term can be onboarded for the first time, or re-onboarded after it has been offboarded.
+contract LendingTermOnboarding is VoltGovernor {
```

Comment:
it feels a bit strange to have multiple governor DAO's and each of them have the governor role, even though all this can do is onboard collateral.

maybe it makes sense to modify the governor in a way that it can make calls to trusted addresses, present a unique identifier, receive arbitrary calldata, and then propose that calldata to the timelock. then, this contract doesn't have to be privileged and is simply a factory + calldata generator.

Reply:
OTOH, that would be a lift, and LendingTermOffboarding needs the governor role to quickly off-board terms in case of emergency.

Reply:
that pattern you suggest reminds me of the Fei Protocol sentinel/guard architecture https://github.com/fei-protocol/fei-protocol-core/tree/develop/contracts/sentinel

why not but I really don't mind having the governor role on a bunch of immutable contracts that automate certain governance processes. The effect is the same, the governance process automation contracts can only generate a limited set of calldata, known in advance & testable, so whether they pass these to a privileged generic caller or do the call themselves, I think the risk is the same

Reply:
btw this onboarder is a Governor so it just has the `TIMELOCK_PROPOSER` role and queues the actions in the timelock. The timelock has the `GAUGE_ADD` & `GOVERNOR` roles (and CREDIT can veto the onboarding while it is waiting in the timelock).

Reply:
you're right, this works